### PR TITLE
linux-kernel-starfive-jh7110: new, 6.6.20+vf2sdk5.12.0

### PR DESCRIPTION
**Pull request by Errant404**
Topic Description
-----------------

- linux-kernel-starfive-jh7110: new, 6.6.20+vf2sdk5.12.0
- kernel-build-common: add support for flexible versioning

Package(s) Affected
-------------------

- kernel-build-common: 15
- linux+kernel+starfive+jh7110: 6.6.20+vf2sdk5.12.0
- linux-kernel-starfive-jh7110-6.6.20: 6.6.20+vf2sdk5.12.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit kernel-build-common linux-kernel-starfive-jh7110 linux+kernel+starfive+jh7110
```

Test Build(s) Done
------------------

**Secondary Architectures**

- [x] RISC-V 64-bit `riscv64`

### CODE REVIEW

File: runtime-kernel/linux+kernel+starfive+jh7110/autobuild/build [hunk 1]
```diff
@@ -0,0 +1 @@
+mkdir "$PKGDIR"
```
Comment (MingcongBai) [inline runtime-kernel/linux+kernel+starfive+jh7110/autobuild/build]: Use ABTYPE=dummy instead.

File: runtime-kernel/linux-kernel-starfive-jh7110/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,10 @@
+PKGNAME=linux-kernel-starfive-jh7110-6.6.20
+PKGSEC=kernel
+PKGDEP=""
+BUILDDEP="bc git pahole kernel-build-common"
+PKGDES="Linux Kernel for AOSC OS (for starfive jh7110 devices)"
```
Comment (MingcongBai) [inline runtime-kernel/linux-kernel-starfive-jh7110/autobuild/defines]: for StarFive devices based on the JH7110 SoC

File: runtime-kernel/linux-kernel-starfive-jh7110/spec [hunk 3]
```diff
@@ -0,0 +1,7 @@
+KERNEL_VER=6.6.20
+VF2_SDK_VER=5.12.0
+
+VER=${KERNEL_VER}+vf2sdk${VF2_SDK_VER}
+
+SRCS="git::commit=tags/JH7110_VF2_6.6_v${VF2_SDK_VER}::https://github.com/starfive-tech/linux/"
```
Comment (MingcongBai) [inline runtime-kernel/linux-kernel-starfive-jh7110/spec]: Replace `_6.6_` with `${UPSTREAM_VER%.*}`.

File: runtime-kernel/linux-kernel-starfive-jh7110/spec [hunk 4]
```diff
@@ -0,0 +1,7 @@
+KERNEL_VER=6.6.20
```
Comment (MingcongBai) [inline runtime-kernel/linux-kernel-starfive-jh7110/spec]: Replace `KERNEL_VER` with `UPSTREAM_VER`.

Comment (MingcongBai) [reply runtime-kernel/linux-kernel-starfive-jh7110/spec]: Also, how do we know that it's 6.6.20?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: runtime-kernel/linux-kernel-starfive-jh7110/spec (hunk 4 above)
Comment (Errant404) [reply runtime-kernel/linux-kernel-starfive-jh7110/spec]: You can find out the kernel version from the [Makefile](https://github.com/starfive-tech/linux/blob/JH7110_VF2_6.6_v5.12.0/Makefile) in their official repository.

Comment (MingcongBai) [reply runtime-kernel/linux-kernel-starfive-jh7110/spec]: Then I would add a comment explaining it in `spec`, right above `UPSTREAM_VER`.

Comment (Errant404) [reply runtime-kernel/linux-kernel-starfive-jh7110/spec]: ~The kernel version is written in the source repository's Makefile. Isn't this supposed to be common knowledge?~
I've added a comment, and you can check if it meets your requirement.

**Review by MingcongBai — APPROVED** [review]
LGTM, tests pending by @Errant404.

**Comment by Errant404** [comment]
The built kernel boots up successfully with the IMG PowerVR GPU driver loaded correctly.
图片

**Pull request merged**